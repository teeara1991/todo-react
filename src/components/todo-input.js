import React from "react";
export function TodoInput(props){
  const onSubmit = event => {
    event.preventDefault();
    const todoName = event.target.todo_name
    props.onSubmit({data: todoName.value});
    todoName.value = "";
    todoName.focus();
  };
    return (
      <form name="todo" onSubmit={onSubmit}>
            <div className="input-group mb-3">
              <input
                type="text"
                name="todo_name"
                className="form-control"
                placeholder="Добавить задачу"
                aria-describedby="basic-addon2"
                autoFocus/>
              <div className="input-group-append">
                <button
                  type="submit"
                  className="btn btn-primary btn-outline-secondary">
                  +
                </button>
              </div>
            </div>
          </form>
    );
  }