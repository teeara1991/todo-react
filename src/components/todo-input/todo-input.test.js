import React from "react";
import { TodoInput } from "../todo-input/todo-input";
import { configure, shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

describe("<TodoInput>", () => {
  it("calls AddTodo Redux action creator with button click and form submit", () => {
    const props = {
      onSubmit: jest.fn(),
      todos: []
    };
    const wrapper = shallow(<TodoInput {...props} />);
    wrapper.find("form").simulate("submit", {
      preventDefault: () => {},
      target: { todo_name: { value: "8", focus: jest.fn() } }
    });
    wrapper.find("button").simulate("submit", {
      preventDefault: () => {},
      target: { todo_name: { value: "8", focus: jest.fn() } }
    });
    expect(props.onSubmit).toHaveBeenCalledWith("8");
  });
});
it("matches the snapshot", () => {
  const tree = shallow(<TodoInput />);
  expect(toJson(tree)).toMatchSnapshot();
});
