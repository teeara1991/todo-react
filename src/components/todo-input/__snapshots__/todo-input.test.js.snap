// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<form
  name="todo"
  onSubmit={[Function]}
>
  <div
    className="input-group mb-3"
  >
    <input
      aria-describedby="basic-addon2"
      autoFocus={true}
      className="form-control"
      name="todo_name"
      placeholder="Добавить задачу"
      type="text"
    />
    <div
      className="input-group-append"
    >
      <button
        className="btn btn-primary btn-outline-secondary"
        type="submit"
      >
        +
      </button>
    </div>
  </div>
</form>
`;
